bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -818,8 +828,18 @@ impl<'a> CircuitInputStateRef<'a> {
             self.handle_reversion();
         }
 
-        self.tx_ctx.pop_call_ctx();
+        // If current call has caller.
+        // TODO: use is_root??
```

Comment:
If `is_root == true` the "last" return data size will be `0` so will always revert if `size > 0`? Not really sure what the behavior is then when RETURNDATACOPY is called with `size == 0`? If this is an edge case/special case probably good to document in the spec as well.

Reply:
> If is_root == true the "last" return data size will be 0 so will always revert if size > 0? 
From geth code, evm checks len(returnData) as boundary, which is 0 ( len(nil) ) if there is no previous call. so in this case, no matter is_root or not, all non-zero length copy attempts is reverted due to out-of-bound error, but 0 length should be ok. 
Here is_root checking is just to see if we need to update caller's laster_callee_xxx value. As root call has no caller, and I think there is no way to run RETURNDATACOPY after root func returns, so in Err(_) case I did nothing.

Reply:
Okay I see, that makes sense. I also see that `LastCalleeReturnDataOffse` and `LastCalleeReturnDataLength` are forced to `0` in `call` so looks good to me.